substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -1860,6 +1941,41 @@ impl<T: Config> StakingInterface for Pallet<T> {
 	}
 }
 
+impl<T: Config> sp_staking::StakingUnsafe for Pallet<T> {
+	fn migrate_to_virtual_staker(who: &Self::AccountId) {
+		T::Currency::remove_lock(crate::STAKING_ID, who);
+		VirtualStakers::<T>::insert(who, ());
+	}
+
+	fn virtual_bond(
+		who: &Self::AccountId,
+		value: Self::Balance,
+		payee: &Self::AccountId,
```

Comment:
As per the comments above, `RewardDestination<AccountId>` instead of accountid would be great for generality. However, in this specific case it's not so important, since the payee of a virtual bond will always have to be `Stash(accountid)` (where accountId != who). This assumption could change in the future though.

Reply:
I agree it can be. Though not sure where the boundary is to overgeneralizing 😀 . Nobody else uses this `StakingInterface` and we don't have that requirement. This will only currently disable self compounding.

Reply:
As stated above, I wouldn't introduce `RewardDestination` here because, in short:
1. it's an implementation detail of the higher impl type
2. we don't have a use case for another destination other than an account